CI triage:
- Travis CI hit https://github.com/cilium/cilium/issues/25272
- ConformanceGatewayAPI / gateway-api-conformance-test failed with something new seemingly

```
=== RUN   TestConformance/TLSRouteSimpleSameNamespace
    suite.go:258: Applying tests/tlsroute-simple-same-namespace.yaml
    apply.go:211: Creating gateway-conformance-infra-test TLSRoute
    apply.go:211: Creating gateway-tlsroute Gateway
    helpers.go:290: Gateway expected observedGeneration to be updated for all conditions, only 0/1 were updated. stale conditions are: Accepted
    helpers.go:441: Accepted condition set to Status False with Reason InvalidTLSRoute, expected Status True
    helpers.go:441: Accepted was not in conditions list [[{Accepted False 1 2023-05-10 12:36:01 +0000 UTC InvalidTLSRoute Gateway.gateway.networking.k8s.io "gateway-tlsroute" not found}]]
    helpers.go:441: Accepted condition set to Status False with Reason InvalidTLSRoute, expected Status True
    helpers.go:441: Accepted was not in conditions list [[{Accepted False 1 2023-05-10 12:36:01 +0000 UTC InvalidTLSRoute Gateway.gateway.networking.k8s.io "gateway-tlsroute" not found}]]
    helpers.go:441: Accepted condition set to Status False with Reason InvalidTLSRoute, expected Status True
    [SNIP, lots of repetition]
    helpers.go:441: Accepted was not in conditions list [[{Accepted False 1 2023-05-10 12:36:01 +0000 UTC InvalidTLSRoute Gateway.gateway.networking.k8s.io "gateway-tlsroute" not found}]]
    helpers.go:441: Accepted condition set to Status False with Reason InvalidTLSRoute, expected Status True
    helpers.go:441: Accepted was not in conditions list [[{Accepted False 1 2023-05-10 12:36:01 +0000 UTC InvalidTLSRoute Gateway.gateway.networking.k8s.io "gateway-tlsroute" not found}]]
    tlsroute-simple-same-namespace.go:51: 
        	Error Trace:	/home/runner/work/cilium/cilium/vendor/sigs.k8s.io/gateway-api/conformance/utils/kubernetes/helpers.go:445
        	            				/home/runner/work/cilium/cilium/vendor/sigs.k8s.io/gateway-api/conformance/utils/kubernetes/helpers.go:606
        	            				/home/runner/work/cilium/cilium/vendor/sigs.k8s.io/gateway-api/conformance/tests/tlsroute-simple-same-namespace.go:51
        	            				/home/runner/work/cilium/cilium/vendor/sigs.k8s.io/gateway-api/conformance/utils/suite/suite.go:262
        	            				/home/runner/work/cilium/cilium/vendor/sigs.k8s.io/gateway-api/conformance/utils/suite/suite.go:220
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestConformance/TLSRouteSimpleSameNamespace
        	Messages:   	error waiting for TLSRoute to have parents matching expectations
    apply.go:219: Deleting gateway-tlsroute Gateway
    apply.go:219: Deleting gateway-conformance-infra-test TLSRoute


     --- FAIL: TestConformance/TLSRouteSimpleSameNamespace (61.07s)
FAIL
FAIL	github.com/cilium/cilium/operator/pkg/gateway-api	117.095s
FAIL
```

rerunning to see if it's flaky or just broken by me

--- a/pkg/allocator/allocator.go
+++ b/pkg/allocator/allocator.go
@@ -88,7 +88,7 @@ const (
 type Allocator struct {
 	// events is a channel which will receive AllocatorEvent as IDs are
 	// added, modified or removed from the allocator
-	events AllocatorEventChan
+	events AllocatorEventSendChan
 
 	// keyType is an instance of the type to be used as allocator key.
 	keyType AllocatorKey
@@ -345,7 +345,7 @@ func WithBackend(backend Backend) AllocatorOption {
 // read while NewAllocator() is being called to ensure that the channel does
 // not block indefinitely while NewAllocator() emits events on it while
 // populating the initial cache.
-func WithEvents(events AllocatorEventChan) AllocatorOption {
+func WithEvents(events AllocatorEventSendChan) AllocatorOption {
 	return func(a *Allocator) { a.events = events }
 }
 
@@ -381,7 +381,7 @@ func WithoutGC() AllocatorOption {
 // GetEvents returns the events channel given to the allocator when
 // constructed.
 // Note: This channel is not owned by the allocator!
-func (a *Allocator) GetEvents() AllocatorEventChan {
+func (a *Allocator) GetEvents() AllocatorEventSendChan {
 	return a.events
 }
 
@@ -866,6 +866,10 @@ func (a *Allocator) startLocalKeySync() {
 
 // AllocatorEventChan is a channel to receive allocator events on
 type AllocatorEventChan chan AllocatorEvent
+
+// Send- and receive-only versions of the above.
+type AllocatorEventRecvChan = <-chan AllocatorEvent
+type AllocatorEventSendChan = chan<- AllocatorEvent
 
 // AllocatorEvent is an event sent over AllocatorEventChan
 type AllocatorEvent struct {

--- a/pkg/identity/cache/allocator.go
+++ b/pkg/identity/cache/allocator.go
@@ -49,6 +49,9 @@ type CachingIdentityAllocator struct {
 
 	identitiesPath string
 
+	// This field exists is to hand out references that are either for sending
+	// and receiving. It should not be used directly without converting it first
+	// to a AllocatorEventSendChan or AllocatorEventRecvChan.
 	events  allocator.AllocatorEventChan
 	watcher identityWatcher
 
@@ -167,7 +170,7 @@ func (m *CachingIdentityAllocator) InitIdentityAllocator(client clientset.Interf
 
 	// Asynchronously set up the global identity allocator since it connects
 	// to the kvstore.
-	go func(owner IdentityAllocatorOwner, events allocator.AllocatorEventChan, minID, maxID idpool.ID) {
+	go func(owner IdentityAllocatorOwner, events allocator.AllocatorEventSendChan, minID, maxID idpool.ID) {
 		m.setupMutex.Lock()
 		defer m.setupMutex.Unlock()
 
@@ -273,7 +276,9 @@ func (m *CachingIdentityAllocator) Close() {
 
 	m.IdentityAllocator.Delete()
 	if m.events != nil {
-		close(m.events)
+		// Have the now only remaining writing party close the events channel,
+		// to ensure we don't panic with 'send on closed channel'.
+		m.localIdentities.close()
 		m.events = nil
 	}
 

--- a/pkg/identity/cache/cache.go
+++ b/pkg/identity/cache/cache.go
@@ -134,7 +134,7 @@ func collectEvent(event allocator.AllocatorEvent, added, deleted IdentityCache) 
 }
 
 // watch starts the identity watcher
-func (w *identityWatcher) watch(events allocator.AllocatorEventChan) {
+func (w *identityWatcher) watch(events allocator.AllocatorEventRecvChan) {
 
 	go func() {
 		for {

--- a/pkg/identity/cache/local.go
+++ b/pkg/identity/cache/local.go
@@ -22,10 +22,10 @@ type localIdentityCache struct {
 	nextNumericIdentity identity.NumericIdentity
 	minID               identity.NumericIdentity
 	maxID               identity.NumericIdentity
-	events              allocator.AllocatorEventChan
+	events              allocator.AllocatorEventSendChan
 }
 
-func newLocalIdentityCache(minID, maxID identity.NumericIdentity, events allocator.AllocatorEventChan) *localIdentityCache {
+func newLocalIdentityCache(minID, maxID identity.NumericIdentity, events allocator.AllocatorEventSendChan) *localIdentityCache {
 	return &localIdentityCache{
 		identitiesByID:      map[identity.NumericIdentity]*identity.Identity{},
 		identitiesByLabels:  map[string]*identity.Identity{},
@@ -190,4 +190,16 @@ func (l *localIdentityCache) GetIdentities() map[identity.NumericIdentity]*ident
 	}
 
 	return cache
+}
+
+// close closes the events channel. The local identity cache is the writing
+// party, hence also needs to close the channel.
+func (l *localIdentityCache) close() {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+
+	if l.events != nil {
+		close(l.events)
+		l.events = nil
+	}
 }

